src/renderer/services/midgard/pools.ts

```diff
@@ -689,6 +692,47 @@ const createPoolsService = (
       RxOp.shareReplay(1)
     )
 
+  // Factory to get depth history from Midgard
+  const apiGetDepthHistory$ = (params: ApiGetDepthHistoryParams): DepthHistoryLD => {
+    const { poolAsset, ...otherParams } = params
+    return FP.pipe(
+      midgardDefaultApi$,
+      liveData.chain((api) =>
+        FP.pipe(
+          api.getDepthHistory({ pool: assetToString(poolAsset), ...otherParams }),
+          RxOp.map(RD.success),
+          RxOp.startWith(RD.pending),
+          RxOp.catchError((e: Error) => Rx.of(RD.failure(e)))
+        )
+      ),
+      RxOp.shareReplay(1)
+    )
+  }
+
+  const { stream$: reloadDepthHistory$, trigger: reloadDepthHistory } = triggerStream()
+
+  const getDepthHistory$ = (params: GetDepthHistoryParams): DepthHistoryLD =>
+    FP.pipe(
+      Rx.combineLatest([selectedPoolAsset$, reloadDepthHistory$]),
+      RxOp.filter(([oSelectedPoolAsset, _]) => O.isSome(oSelectedPoolAsset)),
```

Comment:
Pls remove this line ^ . _I know we have it similar at other places, but @thatStrangeGuyThorchain has realized today, it breaks data flow in a stream and we already handle the case of `O.none` in line 723.